### ISSUE TITLE
Proposal: Pass config variables as build arguments into Docker builds

### ISSUE THREAD

**Issue by jayjun** [deis/builder#479]
As discussed in #450, config variables are passed as environment variables during build-time to buildpacks, but the same doesn't happen for Dockerfile deploys. In other words, we can't parameterise Docker builds.

The [ARG](https://github.com/docker/docker/commit/54240f8da9992880e20a1508e9a6f0e59f2adef1) primitive and `--build-arg` flag were introduced to pass build-time environment variables.

```Bash
$ deis config:set CREDS=s3cr3t
```
```Dockerfile
ARG CREDS
RUN echo $CREDS > creds.txt && proxy -creds=creds.txt
```
I propose mapping config variables to Docker build-time variables to mirror buildpack behaviour.

**Comment by bacongobbler** [comment]
See https://github.com/deis/builder/pull/478 and https://github.com/deis/dockerbuilder/pull/114.

**Comment by Akshaykapoor** [comment]
It'll be good to have this. @bacongobbler when do you think it can be merged

**Comment by bacongobbler** [comment]
I don't have time to bring them past the finish line. Feel free to take it up!

**Comment by ApsOps** [comment]
Should be fixed by #478 , #488 and https://github.com/deis/dockerbuilder/pull/114, https://github.com/deis/dockerbuilder/pull/117
